monitoring/prober/scd/test_operation_simple_heavy_traffic_concurrent.py

```diff
@@ -0,0 +1,365 @@
+# The operation lasts 9 mins and the time window is one after one per sequential idx change
+# The 2D area and altitude won't change with idx
+def _make_op_request_differ_in_time(idx):
+  delta = 10
+  time_start = datetime.datetime.utcnow() + datetime.timedelta(minutes=20) + datetime.timedelta(minutes=delta * idx)
+  time_end = time_start + datetime.timedelta(minutes=delta - 1)
+
+  vol4 = common.make_vol4(time_start, time_end, 0, 120, common.make_circle(-56, 178, 50))
+  return _make_op_request_with_extents(vol4)
+
+
+# Generate request with non-overlapping operations in volume4d.
+# 1/3 operations will be generated with different 2d areas, altitude ranges and time windows respectively
+def _make_op_request(idx):
+  if idx < GROUP_SIZE:
+    return _make_op_request_differ_in_2d(idx)
+  elif idx < GROUP_SIZE * 2:
+    return _make_op_request_differ_in_altitude(idx)
+  else:
+    return _make_op_request_differ_in_time(idx)
+
+
+def _intersection(list1, list2):
+  return list(set(list1) & set(list2))
+
+
+def _put_operation(req, op_id, scd_session):
+  return scd_session.put('/operation_references/{}'.format(op_id), json=req, scope=SCOPE_SC)
+
+
+def _get_operation(op_id, scd_session):
+  return scd_session.get('/operation_references/{}'.format(op_id), scope=SCOPE_SC)
+
+
+def _query_operation(idx, scd_session):
+  lat = _calculate_lat(idx)
+  return scd_session.post('/operation_references/query', json={
+    'area_of_interest': common.make_vol4(None, None, 0, 5000, common.make_circle(lat, 178, 12000))
+  }, scope=SCOPE_SC)
+
+
+def _build_mutate_request(idx, op_id, op_map, scd_session):
+  # GET current op
+  resp = scd_session.get('/operation_references/{}'.format(op_id))
+  assert resp.status_code == 200, resp.content
+  existing_op = resp.json().get('operation_reference', None)
+  assert existing_op is not None
+  op_map[op_id] = existing_op
+
+  req = _make_op_request(idx)
+  req = {
+    'key': [existing_op["ovn"]],
+    'extents': req['extents'],
+    'old_version': existing_op['version'],
+    'state': 'Activated',
+    'uss_base_url': 'https://example.com/uss2',
+    'subscription_id': existing_op['subscription_id']
+  }
+  return req
+
+
+def _delete_operation(op_id, scd_session):
+  return scd_session.delete('/operation_references/{}'.format(op_id), scope=SCOPE_SC)
+
+
+def _collect_resp_callback(key, op_resp_map, future):
+  op_resp_map[key] = future.result()
+
+
+def test_ensure_clean_workspace(scd_session):
+  for op_id in OP_IDS:
+    resp = scd_session.get('/operation_references/{}'.format(op_id), scope=SCOPE_SC)
+    if resp.status_code == 200:
+      resp = scd_session.delete('/operation_references/{}'.format(op_id), scope=SCOPE_SC)
+      assert resp.status_code == 200, resp.content
+    elif resp.status_code == 404:
+      # As expected.
+      pass
+    else:
+      assert False, resp.content
+
+
+# Preconditions: None
+# Mutations: None
+@default_scope(SCOPE_SC)
```

Comment:
I think we can remove this test and the following one unless they are converted to concurrent requests -- we already have test coverage that verifies non-existent ops are inaccessible via get & query.

Reply:
Gotcha, removed the test cases to verify the deletion since as you mentioned they are already covered in other places.

In terms of the concurrent get/query,  we have them right after the concurrent creation to verify the creation.